llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -688,16 +704,25 @@ bool PostPipeliner::tryHeuristics() {
       continue;
     }
     ConfigStrategy S(*DAG, Info, MinLength + Extra * II, Mode, FromTop, Param);
-    resetSchedule();
-    LLVM_SUMMARY(dbgs() << "--- Strategy " << S.name());
+    resetSchedule(/*ResetCritical=*/true);
+    DEBUG_SUMMARY(dbgs() << "--- Strategy " << S.name());
+    if (scheduleFirstIteration(S) && scheduleOtherIterations()) {
+      DEBUG_SUMMARY(dbgs() << " found II=" << II << "\n");
+      return true;
+    }
+
+    DEBUG_SUMMARY(dbgs() << " failed\n");
+    resetSchedule(/*ResetCritial=*/false);
```

Comment:
Note: we could run multiple times without resetting the critical path to highlight more problematic paths and point out 'multi-critical nodes'. We'd need to accumulate criticality in a sane manner (and probably think about possible overflow of those numbers)